build/makelib/golang.mk

```diff
@@ -195,3 +195,25 @@ $(GOJUNIT):
 	@mkdir -p $(TOOLS_HOST_DIR)/tmp
 	@GOPATH=$(TOOLS_HOST_DIR)/tmp GOBIN=$(TOOLS_HOST_DIR) $(GOHOST) get github.com/jstemmer/go-junit-report
 	@$(GOHOST) clean -modcache
+
+export CONTROLLER_GEN=$(TOOLS_HOST_DIR)/controller-gen-$(CONTROLLER_GEN_VERSION)
+export YQ=$(TOOLS_HOST_DIR)/yq-v3
+$(CONTROLLER_GEN) $(YQ):
+	{ \
+		set -ex ;\
+		mkdir -p $(TOOLS_HOST_DIR) ;\
+		CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
+		cd $$CONTROLLER_GEN_TMP_DIR ;\
+		go mod init tmp;\
+		unset GOOS GOARCH  # avoid error: "cannot install cross-compiled binaries when GOBIN is set" ;\
+		export CGO_ENABLED=0  # do not need gcc nor the errors that come with not having it ;\
+		export GOBIN=$$CONTROLLER_GEN_TMP_DIR  # go get dependencies into the temp dir ;\
+		env | grep -E "^GO*" ;\
```

Comment:
Is it intended? Or just a test?

Reply:
Intended. This changes it so controller-gen isn't installed system-wide (which got out of sync for me) and installs it in the `.cache/tools` dir.

Reply:
I'm still not sure I understand, why printing and grepping the env vars?

Reply:
OH! That was just for my debugging! Will remove!